src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/AggregatedStatisticsAdapter.java

```diff
@@ -21,64 +22,43 @@
 import de.dennisguse.opentracks.settings.UnitSystem;
 import de.dennisguse.opentracks.util.StringUtils;
 
-public class AggregatedStatisticsAdapter extends BaseAdapter {
+public class AggregatedStatisticsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
 
-    private AggregatedStatistics aggregatedStatistics;
     private final Context context;
+    private AggregatedStatistics aggregatedStatistics;
 
     public AggregatedStatisticsAdapter(Context context, AggregatedStatistics aggregatedStatistics) {
         this.context = context;
         this.aggregatedStatistics = aggregatedStatistics;
     }
 
+    @NonNull
     @Override
-    public int getCount() {
-        if (aggregatedStatistics != null) {
-            return aggregatedStatistics.getCount();
-        }
-        return 0;
-    }
-
-    @Override
-    public AggregatedStatistics.AggregatedStatistic getItem(int position) {
-        return aggregatedStatistics.getItem(position);
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.aggregated_stats_list_item, parent, false);
+        return new ViewHolder(view);
     }
 
     @Override
-    public long getItemId(int position) {
-        // Not important because data are not database register but cooked data.
-        return position;
-    }
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+        ViewHolder viewHolder = (ViewHolder) holder;
 
-    @Override
-    public View getView(int position, View convertView, ViewGroup parent) {
-        AggregatedStatistics.AggregatedStatistic aggregatedStatistic = getItem(position);
-
-        ViewHolder viewHolder;
-        if (convertView == null) {
-            convertView = LayoutInflater.from(context).inflate(R.layout.aggregated_stats_list_item, parent, false);
-            viewHolder = new ViewHolder(convertView);
-            convertView.setTag(viewHolder);
-        } else {
-            viewHolder = (ViewHolder) convertView.getTag();
-        }
+        AggregatedStatistics.AggregatedStatistic aggregatedStatistic = aggregatedStatistics.getItem(position);
 
-        Resources resources = context.getResources();
-        String localizedActivityType = aggregatedStatistic.getActivityTypeLocalized();
-        if (ActivityType.findByLocalizedString(resources, localizedActivityType)
-                .isShowSpeedPreferred()) {
+        String type = aggregatedStatistic.getActivityTypeLocalized();
+        if (ActivityType.findByLocalizedString(context, type).isShowSpeedPreferred()) {
             viewHolder.setSpeed(aggregatedStatistic);
         } else {
             viewHolder.setPace(aggregatedStatistic);
         }
-
-        return convertView;
     }
 
-    public AggregatedStatistics swapData(AggregatedStatistics aggregatedStatistics) {
```

Comment:
Nitpicking: diff is easier to read when methods are not switched.
Switching getCategories() and swapData() shows that swapData() actually didn't change.

PS/ Could it be that your AndroidStudio as a cleanup?
Also happened for instance variables (context and aggregatedStatistics).